crates/terminal_view/src/terminal_panel.rs

```diff
@@ -266,8 +291,139 @@ impl TerminalPanel {
         };
 
         this.update(cx, |this, cx| {
-            this.add_terminal(Some(action.working_directory.clone()), cx)
+            this.add_terminal(Some(action.working_directory.clone()), None, cx)
+        })
+    }
+
+    pub fn spawn_runnable(
```

Comment:
Nit: this function could probably be made a bit smaller, especially around the `allow_multiple` if/else.